### PR TITLE
PoC to install on demand postcss-markdown dependency

### PR DESCRIPTION
**Pull request by gucong3000**
> Which issue, if any, is this issue related to?

- #3248
- #3264

> Is there anything in the PR that needs further explanation?

Drop built-in support for markdown fences.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,6 +100,7 @@
     "npm-run-all": "^4.0.2",
     "npmpub": "^3.1.0",
     "postcss-import": "^11.0.0",
+    "postcss-markdown": "^0.30.0",
```
Comment (ntwb) [inline package.json]: @gucong3000 Should `postcss-markdown` have moved from `dependencies` to `devDependencies`?

**Comment by jeddy3** [comment]
This is a breaking change.

Let's make sure [we're all happy with the direction we're taking](https://github.com/stylelint/stylelint/issues/3264#issuecomment-399043917) before we merge this.

For example, if we want to go down the route of supporting non-standard things, then we should probably investigate how we can roll-up postcss-markdown into a separate bundle to reduce weight and help with performance.

**Comment by evilebottnawi** [comment]
@jeddy3 I'm against removal `postcss-markdown`, it is help linting a lot of documentation, also i don't see any problems in `postcss-markdown`

**Comment by ntwb** [comment]
@evilebottnawi It was originally proposed to be removed to improve performance. I'd also and I will continue to use it even if I have to install `postcss-markdown` separately lint documentation.

As such per the original decision to remove it for performance reasons is why I approved this PR. Now as noted over in https://github.com/stylelint/stylelint/issues/3264#issuecomment-399043917 we are still considering performance, but we're also considering adding and refactoring stylelint syntaxes supported, if we are going to add those currently suggested then I'm in agreement with you that markdown should also be included.

**Comment by evilebottnawi** [comment]
@ntwb Somebody have benchmark? How we decrease perf using `postcss-markdown` out of box?

**Comment by ntwb** [comment]
@evilebottnawi, apologies, not _performance_ it was to reduce the size of the stylelint package #3248

**Comment by evilebottnawi** [comment]
@ntwb i think `postcss-markdown` it not large :smile:

**Comment by gucong3000** [comment]
I pushed a new commit to implement such a feature:
```bash
$ stylelint "**/*.md"
PostCSS syntax module missing: "postcss-markdown"
Please use "--fix" flag for automatic install or install it by yourself

$ stylelint "**/*.md" --fix
Automatic installation missing module: postcss-markdown
+ postcss-markdown@0.30.0
added 1 package from 1 contributor and audited 25185 packages in 32.202s

lib/__tests__/fixtures/style-tag.markdown
 6:1  ×  Unexpected empty block   block-no-empty
```

**Comment by jeddy3** [comment]
> I pushed a new commit to implement such a feature:

Thanks for this PoC. It looks interesting. I think we should hold off on the idea for now and maybe roll it out at a later date as, I believe, this is a breaking change.

We can add CSS-in-JS support in as a non-breaking change. So, let's do that.

**Pull request closed without merging**